diff --git a/build/cli.js b/build/cli.js
index 48c815138..e69fa024a 100644
--- a/build/cli.js
+++ b/build/cli.js
@@ -1,6 +1,6 @@
 const fs = require("fs");
 const path = require("path");
-
+const { performance } = require("perf_hooks");
 const cliProgress = require("cli-progress");
 
 const { CONTENT_ROOT, Document, slugToFolder } = require("../content");
@@ -52,12 +52,16 @@ async function buildDocuments() {
   }
 
   !options.noProgressbar && progressBar.start(documents.count);
+  let c = 0;
   for (const document of documents.iter()) {
+    let T = `${++c} ${document.url}`;
+
+    var t0 = performance.now();
     const outPath = path.join(BUILD_OUT_ROOT, slugToFolder(document.url));
     fs.mkdirSync(outPath, { recursive: true });
 
     const [builtDocument, liveSamples] = await buildDocument(document);
-
+    var t1 = performance.now();
     fs.writeFileSync(
       path.join(outPath, "index.html"),
       renderHTML(builtDocument, document.url)
@@ -74,6 +78,7 @@ async function buildDocuments() {
       fs.mkdirSync(path.dirname(liveSamplePath), { recursive: true });
       fs.writeFileSync(liveSamplePath, html);
     }
+    var t2 = performance.now();
 
     // Collect non-archived documents' slugs to be used in sitemap building and
     // search index building
@@ -90,8 +95,11 @@ async function buildDocuments() {
     if (!options.noProgressbar) {
       progressBar.increment();
     } else {
-      console.log(outPath);
+      // console.log(outPath);
     }
+    var t3 = performance.now();
+    console.log(T, JSON.stringify([t3 - t2, t2 - t1, t1 - t0]));
+    // console.timeEnd(T);
   }
 
   !options.noProgressbar && progressBar.stop();
